ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs

```diff
@@ -719,12 +730,6 @@ equatingEither eqA _ (Left  x) (Left  y) = eqA x y
 equatingEither _ eqB (Right x) (Right y) = eqB x y
 equatingEither _ _   (Right _) (Left  _) = False
 
-comparingFst :: Comparing a -> Comparing (a, b)
-comparingFst cmp = cmp `on` fst
-
-equatingFst :: Equating a -> Equating (a, b)
-equatingFst eq = eq `on` fst
-
```

Comment:
The removal of `comparingFst` and `equatingFst` and the cascade of changes seems a bit heavy-handed, I admit. I'm open to suggestions -- I suspect the related function's implementations could benefit from some newtypes.